scheduler/executor/interfaces/snapshot_readerwriter.go

```diff
@@ -0,0 +1,11 @@
+package interfaces
+
+import "io"
+
+// SnapshotReaderWriter provides an interface to read and write the snapshot
+type SnapshotReaderWriter interface {
```

Comment:
Lets create a new folder called `snapshot`
and 2 interfaces
`Reader`
`Writer`.

so usage should be snapshot.Reader
snapshot.Writer

If you want to use both you can always right. This is like io.Writer, io.Reader. single method interfaces are beautiful